packages/editor-ui/src/components/canvas/Canvas.vue

```diff
@@ -48,10 +65,72 @@ const props = withDefaults(
 	},
 );
 
-const { getSelectedEdges, getSelectedNodes, viewportRef, fitView, project } = useVueFlow({
-	id: props.id,
+const {
+	getSelectedNodes,
+	addSelectedNodes,
+	removeSelectedNodes,
+	viewportRef,
+	fitView,
+	project,
+	nodes: graphNodes,
+} = useVueFlow({ id: props.id, deleteKeyCode: null });
+
+useKeybindings({
+	ctrl_c: () => {
+		if (hasSelection.value) {
+			emit('copy:nodes', selectedNodeIds.value);
+		}
+	},
+	ctrl_x: () => {
+		if (hasSelection.value) {
+			emit('cut:nodes', selectedNodeIds.value);
+		}
+	},
+	'delete|backspace': () => {
+		if (hasSelection.value) {
+			emit('delete:nodes', selectedNodeIds.value);
+		}
+	},
+	ctrl_d: () => {
+		if (hasSelection.value) {
+			emit('duplicate:nodes', selectedNodeIds.value);
+		}
+	},
+	d: () => {
+		if (hasSelection.value) {
+			emit('update:nodes:enabled', selectedNodeIds.value);
+		}
+	},
+	p: () => {
+		if (hasSelection.value) {
+			emit('update:nodes:pin', selectedNodeIds.value, 'keyboard-shortcut');
+		}
+	},
+	enter: () => {
+		if (!lastSelectedNode.value) return;
+		emit('update:node:active', lastSelectedNode.value.id);
+	},
+	f2: () => {
+		if (!lastSelectedNode.value) return;
+		emit('update:node:name', lastSelectedNode.value.id);
+	},
+	tab: () => emit('create:node', 'tab'),
+	shift_s: () => emit('create:sticky'),
+	ctrl_alt_n: () => emit('create:workflow'),
+	ctrl_enter: () => emit('run:workflow'),
```

Comment:
I was not sure if this belongs here or in NodeView, I think maybe some shortcuts belong here and other should be moved to NodeView (for example `create:workflow` and `run:workflow`)